src/main/scala/com/iheart/workpipeline/akka/patterns/WorkPipeline.scala

```diff
@@ -131,14 +141,19 @@ case class PullingWorkPipeline( name: String,
                                 iterator: Iterator[_],
                                 pipelineSettings: WorkPipeline.Settings,
                                 backendProps: Props,
+                                metricsCollector: MetricsCollector = NoOpMetricsCollector,
                                 resultChecker: ResultChecker) extends WorkPipeline {
 
-  protected def queueProps = QueueOfIterator.props(iterator, WorkSettings())
+  protected def queueProps = QueueOfIterator.props(iterator, WorkSettings(), metricsCollector)
 
 }
 
 object PullingWorkPipeline {
-  def props(name: String, iterator: Iterator[_], settings: WorkPipeline.Settings, backendProps: Props)
-           (resultChecker: ResultChecker) = Props(PullingWorkPipeline(name, iterator, settings, backendProps, resultChecker))
+  def props(name: String,
+            iterator: Iterator[_],
+            settings: WorkPipeline.Settings,
+            backendProps: Props,
+            metricsCollector: MetricsCollector = NoOpMetricsCollector)(resultChecker: ResultChecker) =
+    Props(PullingWorkPipeline(name, iterator, settings, backendProps, metricsCollector, resultChecker))
 }
```

Comment:
note to self:
* add config wrapper to deal with optional config params
* have prefix be `workPipeline.{name of pipeline}` by default (`workPipeline` part configurable)